app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,58 @@
+class PasswordResetsController < ApplicationController
+  before_action :load_user, only: %i(edit update)
+  before_action :valid_user, only: %i(edit update)
+  before_action :check_expiration, only: %i(edit update)
+
+  def new; end
+
+  def create
+    @user = User.find_by email: params[:password_reset][:email].downcase
+    if @user
+      @user.create_reset_digest
+      @user.send_password_reset_email
+      flash[:info] = t "app.password_resets_controller.create_info"
+      redirect_to root_url
+    else
+      flash.now[:danger] = t "app.password_resets_controller.create_info"
+      render :new
+    end
+  end
+
+  def edit; end
+
+  def update
+    if params[:user][:password].empty?
+      @user.errors.add :password, t("app.password_resets_controller.update_content")
+      render :edit
+    elsif @user.update_attributes(user_params)
+      log_in @user
+      @user.update_attribute(:reset_digest, nil)
+      flash[:success] = t "app.password_resets_controller.update_success"
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit :password, :password_confirmation
+   end
+
+  def load_user
+    @user = User.find_by email: params[:email]
```

Comment:
Không tìm thấy user thì sao ??

Reply:
do trường params[:email] nằm trong link reset password mà link reset password chưa email tồn tại trong database nên mình sẽ luôn load được user, lúc reset password nhập email không tồn tại thì sẽ thông báo "Email address not found", link reset được gởi khi mình nhập email đúng.

Reply:
chuyển hướng đâu không thấy

Reply:
dạ khi mình bấm submit chỗ reset password thì nó sẽ tới load_user, tìm xong nó lưu trong biến @user rồi chuyển tới method valid_user kiểm tra các điều kiện trong method nếu sai thì nó chuyển về trang chủ